rdr_service/api/nph_intake_api.py

```diff
@@ -248,6 +250,18 @@ def post(self):
 
                 all_event_objs.extend(event_objs)
 
+                if activity_data.name in ('consent','withdrawal','deactivate'):
+                    data = {
+                        'event_type': activity_data.name,
+                        'participant_id': participant_id,
+                        'event_authored_time': event_objs[0]['event_authored_time']
+                    }
+                    if GAE_PROJECT == 'localhost':
+                        pass
+                    else:
+                        _task = GCPCloudTask()
+                        _task.execute(endpoint='update_participant_summary_task', payload=data, queue='nph')
+
```

Comment:
@brianmkimmel for this logic, I think it needs to be moved into some sort of data structure and executed **_after data insertions occur_**. 

The reason for this is that the intake API code executes and runs validations on the list of Bundles then builds list of dicts for data inserts so that if some of the payload raises a `Badrequest` then so we don't have discrepancies in the data. We are already having issues with the test payload we have been receiving so this is why I refactored this to work with validation first, then data insertions. 

Since you have this cloud task here there could be inconsistent data with what is getting stored in the participant summary and what we have in the intake events.